CMakeLists.txt

```diff
@@ -34,6 +34,37 @@ endif()
 
 find_ros1_package(std_msgs REQUIRED)
 
+# find all known ROS1 message/service packages
+execute_process(
+    COMMAND "rosmsg" "list"
+    OUTPUT_VARIABLE rosmsg_output
+    ERROR_VARIABLE rosmsg_error
+)
+execute_process(
+    COMMAND "rossrv" "list"
+    OUTPUT_VARIABLE rossrv_output
+    ERROR_VARIABLE rossrv_error
+)
+if(NOT rosmsg_error STREQUAL "" OR NOT rossrv_error STREQUAL "")
+  message(FATAL_ERROR "${rosmsg_error}\n${rossrv_error}\nFailed to find ROS 1 message packages\nPlease make sure to source your ROS2 workspace BEFORE sourcing your ROS1 workspace")
```

Comment:
The reason here is that ROS1 and ROS2 message packages have been found. But the ros2 ones are the first ones on the python path. So the invocation of rosmsg / rossrv will fail because these are python2 scripts trying to import python3 (ROS2) messages. I can make the error message more explicit to reflect what happens exactly.

I'm still not sure why the problem didnt appear when I was calling out bash explicitely though ...